### PR TITLE
[Unit Test] Improve penaltylib coverage

### PR DESCRIPTION
**Pull request by haoxu07**
## Summary
- add CPU-only coverage for BatchedRepetitionPenalizer
- cover valid negative frequency/presence penalties and repetition_penalty < 1.0
- cover speculative repeat penalty expansion and cached overlap-style penalty application

## Related
- Part of #20865

## Test Plan
- PYTHONPATH=python .venv/bin/python - <<'PY'
  import runpy
  import sys
  import torch

  def compile_eager(*args, **kwargs):
      def decorator(fn):
          return fn
      return decorator

  torch.compile = compile_eager
  sys.argv = ["test/registered/unit/sampling/test_penaltylib.py"]
  runpy.run_path("test/registered/unit/sampling/test_penaltylib.py", run_name="__main__")
  PY
- git diff --check

Note: direct local execution without the eager torch.compile shim fails in this macOS environment before tests run due to a torch._inductor/Triton import issue.

---
### CI States

Latest PR Test (Base): :x: [Run #26443372374](https://github.com/sgl-project/sglang/actions/runs/26443372374)
Latest PR Test (Extra): :x: [Run #26443372164](https://github.com/sgl-project/sglang/actions/runs/26443372164)

### CODE REVIEW

**Comment by haoxu07** [comment]
Could a maintainer please add the `run-ci` label when convenient? This PR only updates CPU unit tests for `test/registered/unit/sampling/test_penaltylib.py`.
